gen/build_deploy/bash.py

```diff
@@ -701,24 +707,26 @@ def do_create(tag, build_name, reproducible_artifact_path, commit, variant_argum
     """
     # TODO(cmaloney): Build installers in parallel.
     # Variants are sorted for stable ordering.
-    for variant, bootstrap_info in sorted(variant_arguments.items(),
-                                          key=lambda kv: pkgpanda.util.variant_str(kv[0])):
-        with logger.scope("Building installer for variant: ".format(pkgpanda.util.variant_name(variant))):
-            bootstrap_installer_name = '{}installer'.format(pkgpanda.util.variant_prefix(variant))
-            installer_filename = make_installer_docker(variant, all_completes[variant],
-                                                       all_completes[bootstrap_installer_name])
-
-            yield {
-                'channel_path': 'dcos_generate_config.{}sh'.format(pkgpanda.util.variant_prefix(variant)),
-                'local_path': installer_filename
-            }
+    for variant in sorted(variant_arguments.keys(), key=lambda k: pkgpanda.util.variant_str(k)):
+        variant_name = pkgpanda.util.variant_name(variant)
+        bootstrap_installer_name = '{}installer'.format(pkgpanda.util.variant_prefix(variant))
+        bootstrap_util_name = '{}util'.format(pkgpanda.util.variant_prefix(variant))
+        if bootstrap_installer_name not in all_completes:
+            print('WARNING: No installer tree for variant: {}'.format(variant_name))
+        else:
+            with logger.scope("Building installer for variant: {}".format(variant_name)):
 
-    # Build dcos-launch
-    # TODO(cmaloney): This really doesn't belong to here, but it's the best place it fits for now.
-    #                 dcos-launch works many places which aren't bash / on-premise installers.
-    #                 It also isn't dependent on the "reproducible" artifacts at all. Just the commit...
-    with logger.scope("building dcos-launch"):
-        yield {
-            'channel_path': 'dcos-launch',
-            'local_path': make_dcos_launch()
-        }
+                yield {
+                    'channel_path': 'dcos_generate_config.{}sh'.format(pkgpanda.util.variant_prefix(variant)),
+                    'local_path': make_installer_docker(variant, all_completes[variant],
+                                                        all_completes[bootstrap_installer_name])
+                }
+
+        if bootstrap_util_name not in all_completes:
+            print('WARNING: No util tree for variant: {}'.format(variant_name))
+        else:
+            with logger.scope("building dcos-launch for variant: {}".format(variant_name)):
+                yield {
+                    'channel_path': 'dcos-launch-' + variant_name if variant is not None else 'dcos-launch',
```

Comment:
I see now that this is confusing/misleading syntax but it works as I intended:
`('dcos-launch-' + variant_name) if variant is not None else 'dcos-launch'`